src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java

```diff
@@ -0,0 +1,665 @@
+import static uk.gov.hmcts.reform.civil.utils.PartyUtils.populateDQPartyIds;
+import static uk.gov.hmcts.reform.civil.utils.WitnessUtils.addEventAndDateAddedToRespondentWitnesses;
+
+@Component
+public class SetApplicantResponseDeadline implements CaseTask {
+
+    private final Time time;
+    private final DeadlinesCalculator deadlinesCalculator;
+    private final FrcDocumentsUtils frcDocumentsUtils;
+    private final FeatureToggleService toggleService;
+    private final CaseFlagsInitialiser caseFlagsInitialiser;
+    private final IStateFlowEngine stateFlowEngine;
+    private final AssignCategoryId assignCategoryId;
+    private final ObjectMapper objectMapper;
+    private final CoreCaseUserService coreCaseUserService;
+    private final UserService userService;
+    private final LocationReferenceDataService locationRefDataService;
+    private final CourtLocationUtils courtLocationUtils;
+
+    public SetApplicantResponseDeadline(Time time,
+                                        DeadlinesCalculator deadlinesCalculator,
+                                        FrcDocumentsUtils frcDocumentsUtils,
+                                        FeatureToggleService toggleService,
+                                        CaseFlagsInitialiser caseFlagsInitialiser,
+                                        IStateFlowEngine stateFlowEngine,
+                                        AssignCategoryId assignCategoryId,
+                                        ObjectMapper objectMapper,
+                                        CoreCaseUserService coreCaseUserService,
+                                        UserService userService,
+                                        LocationReferenceDataService locationRefDataService,
+                                        CourtLocationUtils courtLocationUtils) {
+        this.time = time;
+        this.deadlinesCalculator = deadlinesCalculator;
+        this.frcDocumentsUtils = frcDocumentsUtils;
+        this.toggleService = toggleService;
+        this.caseFlagsInitialiser = caseFlagsInitialiser;
+        this.stateFlowEngine = stateFlowEngine;
+        this.assignCategoryId = assignCategoryId;
+        this.objectMapper = objectMapper;
+        this.coreCaseUserService = coreCaseUserService;
+        this.userService = userService;
+        this.locationRefDataService = locationRefDataService;
+        this.courtLocationUtils = courtLocationUtils;
+    }
+
+    @Override
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        CaseData.CaseDataBuilder<?, ?> updatedData = updateRespondentAddresses(caseData);
+
+        LocalDateTime responseDate = time.now();
+        AllocatedTrack allocatedTrack = caseData.getAllocatedTrack();
+        LocalDateTime applicant1Deadline = getApplicant1ResponseDeadline(responseDate, allocatedTrack);
+
+        // 1v2 same legal rep - will respond for both and set applicant 1 response deadline
```

Comment:
Avoid adding comments to code. Your code should be clean enough and descriptive enough to not make comments necessary